clar2wasm/src/error_mapping.rs

```diff
@@ -184,15 +206,45 @@ fn from_runtime_error_code(
             // This RuntimeErrorType::UnwrapFailure need to have a proper context.
             Error::Runtime(RuntimeErrorType::UnwrapFailure, Some(Vec::new()))
         }
-        // TODO: UInt(42) value below is just a placeholder.
-        // It should be replaced by the current "thrown-value" when issue #385 is resolved.
-        // Tests that reach this code are currently ignored.
-        ErrorMap::ShortReturnAssertionFailure => Error::ShortReturn(
-            ShortReturnType::AssertionFailed(Value::Response(ResponseData {
-                committed: false,
-                data: Box::new(Value::UInt(42)),
-            })),
-        ),
+        ErrorMap::ShortReturnAssertionFailure => {
+            let val_offset = instance
+                .get_global(&mut store, "runtime-error-value-offset")
+                .and_then(|glob| glob.get(&mut store).i32())
+                .unwrap_or_else(|| {
+                    panic!("Could not find $runtime-error-value-offset global with i32 value")
+                });
+
+            let type_ser_offset = instance
+                .get_global(&mut store, "runtime-error-type-ser-offset")
+                .and_then(|glob| glob.get(&mut store).i32())
+                .unwrap_or_else(|| {
+                    panic!("Could not find $runtime-error-type-ser-offset global with i32 value")
+                });
+
+            let type_ser_len = instance
+                .get_global(&mut store, "runtime-error-type-ser-len")
+                .and_then(|glob| glob.get(&mut store).i32())
+                .unwrap_or_else(|| {
+                    panic!("Could not find $runtime-error-type-ser-len global with i32 value")
+                });
+
+            let memory = instance
+                .get_memory(&mut store, "memory")
+                .unwrap_or_else(|| panic!("Could not find wasm instance memory"));
+
+            let type_ser_str =
+                read_identifier_from_wasm(memory, &mut store, type_ser_offset, type_ser_len)
+                    .unwrap_or_else(|e| panic!("Could not recover stringified type: {e}"));
+
+            let value_ty = signature_from_string(&type_ser_str, *clarity_version, *epoch_id)
+                .unwrap_or_else(|e| panic!("Could not recover thrown value: {e}"));
+
+            let clarity_val =
+                read_from_wasm_indirect(memory, &mut store, &value_ty, val_offset, *epoch_id)
+                    .unwrap_or_else(|e| panic!("Could not read thrown value from memory: {e}"));
```

Comment:
This piece of code is used is used three times, it might be worth extracting it to it own function.

Reply:
Done. Fixed duplicated code.